pytorch_lightning/trainer/predict_loop.py

```diff
@@ -82,18 +111,43 @@ def predict_step(self, batch, batch_idx, dataloader_idx):
 
         self.trainer.call_hook("on_predict_batch_end", predictions, batch, batch_idx, dataloader_idx)
 
-        self._predictions[dataloader_idx].append(predictions)
+        if self.should_store_predictions:
+            self.predictions[dataloader_idx].append(predictions)
+
+    def _store_batch_indices(self, dataloader_idx: int) -> None:
+        batch_sampler = self.trainer.predict_dataloaders[dataloader_idx].batch_sampler
+        if isinstance(batch_sampler, IndexBatchSamplerWrapper):
+            self.batch_indices = batch_sampler.batch_indices
+            if self.should_store_predictions:
+                self.epoch_batch_indices[dataloader_idx].append(batch_sampler.batch_indices)
+
+    def on_predict_start(self):
+        # enable eval mode + no grads
+        self.on_predict_model_eval()
+        self.trainer.lightning_module.zero_grad()
+        self._previous_grad_status = torch.is_grad_enabled()
+        torch.set_grad_enabled(False)
+
+        # hook
+        self.trainer.call_hook("on_predict_start")
+        self.trainer.call_hook("on_predict_epoch_start")
 
     def on_predict_epoch_end(self):
         self.trainer.profiler.describe()
 
-        results = self._predictions
+        results: List[List[Any]] = self.predictions
 
         self.trainer.call_hook("on_predict_epoch_end", results)
 
-        def _convert_to_numpy(v):
-            return v.cpu().numpy()
+        if self.return_predictions:
+            return results[0] if self.num_dataloaders == 1 else results
 
-        results = apply_to_collection(results, torch.Tensor, _convert_to_numpy)
+    def on_predict_end(self):
+        # clean memory
```

Comment:
mind add a comment about where the predictions go?

Reply:
Done !